climada/engine/test/test_impact_calc.py

```diff
@@ -70,6 +70,16 @@ def test_init(self):
         np.testing.assert_array_equal(HAZ.event_id, icalc.hazard.event_id)
         np.testing.assert_array_equal(HAZ.event_name, icalc.hazard.event_name)
 
+        # Test check matrices
```

Comment:
```suggestion
        # Test check matrices
        # Check fraction and intensity have the same shape
        # Check that explicit zeroes are pruned
```

I am a bit confused why the pruning is tested here...

Reply:
I can also check via a mock if `check_matrices` is called, but I recall strong sentiments against such tests 😄

Reply:
I think my confusion comes from the fact that I am uneasy with having the impact class changing the hazard objects. See comment below.